src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -308,7 +322,7 @@ public ProjectInfo CreateProjectInfoForCurrentState()
                     compilationOptions: this.CurrentCompilationOptions,
                     parseOptions: this.CurrentParseOptions,
                     documents: _documents.Values.Select(d => d.GetInitialState()),
-                    metadataReferences: _metadataReferences.Select(r => r.CurrentSnapshot),
+                    metadataReferences: _metadataReferences.Select(r => r.Target.CurrentSnapshot),
```

Comment:
Convince me that it's always safe to just access `Target` in this file without needing to check nulls?

Reply:
ReferenceCountedDisposable is in one of two states:

1. It's not disposed, so Target is non-null.
2. It's disposed, and Target will throw ObjectDisposedException.

https://github.com/dotnet/roslyn/blob/63433fbc11c5efaeef3f451f8f31df3e76e5684a/src/Workspaces/Core/Portable/Utilities/ReferenceCountedDisposable.cs#L114

Reply:
Then how are we confident throughout this file that it's not disposed?

Reply:
The weaselly answer is "as confident as we were before" -- the places where we were disposing before are unchanged.

The more concrete answer is we do disposal in two places:

1. The project itself being removed.
2. `RemoveMetadataReferenceCore`, which is the same helper that removes the entry from `metadataReferences` as well.

Reply:
Thanks!